superset/config.py

```diff
@@ -203,6 +204,9 @@
 # Maximum number of rows returned in the SQL editor
 SQL_MAX_ROW = 1000
 
+# Maximum number of tables/views displayed in the dropdown window in SQL Lab.
+MAX_TABLE_NAMES = 3000
```

Comment:
in views.py, github hides that diff somehow. I'll change the value to 1000. I haven't seen any lag with 3K table names.